cocos/network/WebSocket.cpp

```diff
@@ -722,8 +726,8 @@ void WebSocket::close()
 
     {
         std::unique_lock<std::mutex> lkClose(_closeMutex);
-        _closeCondition.wait(lkClose);
         _closeState = CloseState::SYNC_CLOSED;
+        _closeCondition.notify_one();
```

Comment:
if `onConnectionClosed` lasts for more than 5ms, it may cause crash issue. Since user's onClose will delete websocket instance, then onConnectionClosed logic may still be executed which will cause crash.
```c++
    // Wait 5 milliseconds for onConnectionClosed to exit!
    std::this_thread::sleep_for(std::chrono::milliseconds(5));
    _delegate->onClose(this);
```
In fact, it's not dead lock in `close`, it's probably stop main thread for several seconds since `close` is a synchronous method.
If developer wanna use asynchronous method, he/she has to use `WebSocket::closeAsync` interface.